state-sync/aptos-data-client/src/peer_states.rs

```diff
@@ -370,6 +370,13 @@ impl PeerStates {
         }
     }
 
+    /// Returns metadata for all peers currently connected to the node
+    pub(crate) fn get_connected_peers_and_metadata(
+        &self,
+    ) -> Result<HashMap<PeerNetworkId, PeerMetadata>, Error> {
+        self.peers_and_metadata.get_connected_peers_and_metadata()
```

Comment:
This is the only thing I'm a little iffy on 🤔 Technically, if all you want is the peers and metadata information, we should pass the `PeersAndMetadata` struct to mempool (as originally planned) and not the data client. But, I assume you've done this because you later want access to the more fine-grained synced version info? (Does that make sense?) 😃 

Let's sync offline -- I think there's a couple options we have and we can see which makes the most sense 😄

Reply:
Yup! Thats exactly the reason. Lets sync offline -- I'd like to get your feedback on the finer grained peer info my prototype is using.